node/src/main/java/io/airlift/node/NodeInfo.java

```diff
@@ -140,6 +143,10 @@ public NodeInfo(String environment,
             this.externalAddress = this.internalAddress;
         }
 
+        if (annotationFile != null && annotationString != null) {
+            throw new IllegalArgumentException("Only one of annotationFile or annotationString should be set, but not both.");
```

Comment:
This should be validated in the config class itself:

```
@AssertTrue("Only one of node.annotations or node.annotations-file can be set")
public boolean isConfigurationValid()
{
   return (annotationFile != null) ^ (annotations != null);
}
```

Reply:
What if someone creates NodeInfo on the fly, without going through config? I think constructor is the best place for such precondition checks.